src/lib.rs

```diff
@@ -203,23 +203,60 @@ pub struct Metrics {
     pub strikeout_thickness: f32,
 }
 
-pub trait Rasterize {
-    /// Errors occurring in Rasterize methods.
-    type Err: ::std::error::Error + Send + Sync + 'static;
+/// Errors occuring when using the rasterizer.
+#[derive(Debug)]
+pub enum Error {
+    /// Couldn't find font matching description.
+    FontNotFound(FontDesc),
+
+    /// Tried to get size metrics from a Face that didn't have a size.
+    MissingSizeMetrics,
+
+    MissingGlyph(RasterizedGlyph),
+
+    /// Requested an operation with a FontKey that isn't known to the rasterizer.
+    UnknownFontKey,
+
+    /// Error from platfrom's font system.
+    PlatformError(String),
+}
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        None
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::FontNotFound(font) => write!(f, "Unable to find the font: {:?}", font),
+            Error::MissingGlyph(glyph) => {
+                write!(f, "Unable to find glyph for char {}", glyph.character)
```

Comment:
```suggestion
                write!(f, "no font with character {:?}", glyph.character)
```

Could go above and beyond and even print its value as bytes, but I think that's unnecessary.

Reply:
I decided to go with slightly different formatting here.